controller/apps.go

```diff
@@ -0,0 +1,413 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AppsController implements the apps resource.
+type AppsController struct {
+	*goa.Controller
+	AuthURL string
+	WitURL  string
+}
+
+// NewAppsController creates a apps controller.
+func NewAppsController(service *goa.Service, config *configuration.Registry) *AppsController {
+	return &AppsController{
+		Controller: service.NewController("AppsController"),
+
+		AuthURL: config.GetAuthServiceURL(),
+		//AuthURL: "http://localhost:8089"
+		//AuthURL: "https://auth.prod-preview.openshift.io",
+		//AuthURL: "https://auth.openshift.io",
+
+		// TODO - make this a config variable?
+		//WitURL: "http://localhost:8080"
+		//WitURL: "http://api.prod-preview.openshift.io",
+		WitURL: "http://api.openshift.io",
+	}
+}
+
+func tostring(item interface{}) string {
+	bytes, _ := json.MarshalIndent(item, "", "  ")
+	return string(bytes)
+}
+
+func (c *AppsController) getAndCheckOsioClient(ctx context.Context) (*OsioClient, error) {
+	oc, err := NewOsioClient(ctx, c.WitURL)
```

Comment:
I've cleaned this up (also for NewAuthClient) as suggested.  I retain the getAndCheck() because they now have the responsiblity of grabbing the token from the context, and in the future may cache the clients for the duration of the API call.